octobot_commons/databases/cache_manager.py

```diff
@@ -0,0 +1,178 @@
+        identifier = config_name or self.DEFAULT_CONFIG_IDENTIFIER
+        try:
+            return bool(self.__class__.CACHES.get_node([tentacle_name, exchange_name, symbol, time_frame, identifier]))
+        except event_tree.NodeExistsError:
+            return False
+
+    def get_cache_registered_requirements(self, tentacle_name, exchange_name, symbol, time_frame, config_name=None):
+        identifier = config_name or self.DEFAULT_CONFIG_IDENTIFIER
+        return self.__class__.CACHES.get_node([tentacle_name, exchange_name, symbol, time_frame, identifier]).\
+            node_value.tentacles_requirements
+
+    def get_cache_previous_db_metadata(self, tentacle_name, exchange_name, symbol, time_frame, config_name=None):
+        identifier = config_name or self.DEFAULT_CONFIG_IDENTIFIER
+        return self.__class__.CACHES.get_node([tentacle_name, exchange_name, symbol, time_frame, identifier]).\
+            node_value.previous_db_metadata
+
+    async def clear_cache(self, tentacle_name, exchange_name=None, symbol=None, time_frame=None, config_name=None):
+        try:
+            for cache, _ in self._caches(tentacle_name, exchange_name, symbol, time_frame, config_name):
+                await cache.node_value.clear()
+            return True
+        except event_tree.NodeExistsError:
+            return False
+
+    async def reset_cache(self, tentacle_name, exchange_name, symbol, time_frame, config_name):
+        identifier = config_name or self.DEFAULT_CONFIG_IDENTIFIER
+        cache = self.__class__.CACHES.delete_node([tentacle_name, exchange_name, symbol, time_frame, identifier])
+        await cache.node_value.close()
+
+    async def close_cache(self, tentacle_name, exchange_name=None, symbol=None, time_frame=None, config_name=None,
+                          reset_cache_db_ids=False):
+        try:
+            to_remove_caches = []
+            for cache, identifiers in self._caches(tentacle_name, exchange_name, symbol, time_frame, config_name):
+                await cache.node_value.close()
+                to_remove_caches.append(identifiers)
+            if reset_cache_db_ids:
+                # remove cache from caches to force complete reopen of the cache db
+                # (might be at a different place)
+                for identifier in to_remove_caches:
+                    self.__class__.CACHES.delete_node(identifier)
+            return True
+        except event_tree.NodeExistsError:
+            return False
+
+    async def reset(self):
+        for cache, _ in self._caches(None, None, None, None, None):
+            if cache.node_value.is_open():
+                await cache.node_value.close()
+        self.__class__.CACHES = event_tree.EventTree()
+
+    def _caches(self, tentacle_name, exchange_name, symbol, time_frame, config_name):
+        path = []
+        for element in (tentacle_name, exchange_name, symbol, time_frame, config_name):
+            if element is None:
+                break
```

Comment:
What about a cache with a None symbol but with a time_frame?

Reply:
That's right, I'll add a different default value to allow for None is it might make sense sometimes to have a `None` key